tests/RobotFramework/tests/cumulocity/registration/registration_lifecycle.robot

```diff
@@ -234,6 +254,12 @@ Entities send to cloud on restart
 
 *** Keywords ***
 
+Get Retained Message Count
+    [Arguments]    ${topic}
+    ${output}=    Execute Command    mosquitto_sub --retained-only -W 3 -t "${topic}" -v    exp_exit_code=27    return_stdout=True
+    ${message_count}=    Get Length    ${output.splitlines()}
+    [Return]    ${message_count}
```

Comment:
Should be using `RETURN` over `[Return]` as the latter is marked as deprecated (see https://github.com/robotframework/robotframework/issues/4078).

```suggestion
    RETURN    ${message_count}
```

Reply:
Though the more I look at the `Get Retained Message Count` the more it would suggest doing the assertion within the keyword, as all references do a follow up assertion on the count.

And it could be simplified by using the [Length Should Be](https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Length%20Should%20Be) built-in assertion.

The keyword could be refactored to the following:

```
Should Have Retained Message Count
    [Arguments]    ${topic}    ${exp_count}
    ${output}=    Execute Command    mosquitto_sub --retained-only -W 3 -t "${topic}" -v    exp_exit_code=27    return_stdout=True
    Length Should Be    ${output.splitlines()}    ${exp_count}
```

And calls to the keyword would just require a single call, and you don't have to deal with the returned value:

```
Should Have Retained Message Count    te/device/${CHILD_SN}/+/+/#    0
```